crates/bevy_render/src/mesh/primitives/dim2.rs

```diff
@@ -400,6 +400,55 @@ impl From<CircularSegment> for Mesh {
     }
 }
 
+/// A builder used for creating a [`Mesh`] with a [`ConvexPolygon`] shape.
+pub struct ConvexPolygonMeshBuilder<const N: usize> {
+    pub vertices: [Vec2; N]
+}
+
+impl<const N: usize> Meshable for ConvexPolygon<N> {
+    type Output = ConvexPolygonMeshBuilder<N>;
+
+    fn mesh(&self) -> Self::Output {
+        Self::Output {
+            vertices: self.vertices,
+        }
+    }
+}
+
+impl<const N: usize> MeshBuilder for ConvexPolygonMeshBuilder<N> {
+    fn build(&self) -> Mesh {
+        let mut indices = Vec::with_capacity(self.vertices.len() - 2);
```

Comment:
I scribbled some polygons on paper, and I think the indices should be like this:

```suggestion
        let mut indices = Vec::with_capacity((self.vertices.len() - 2) * 3);
```

This is also what the ellipse mesh builder and therefore circles and regular polygons use.

Reply:
True! Here I was thinking in terms of triangles, but `indices` actually stores all the indices contiguously.